src/main/java/CleanCycle/Analytics/Main.java

```diff
@@ -466,12 +466,23 @@ public void run() {
             }
 
             synchronized (points) {
```

Comment:
@Metasynic  Since this was a quick patch I didn't fully consider this when committing, but could you change the scope of the lock on points here when you refactor so that the thread doesn't keep hold of the lock while it is sleeping on the endpoint connection - this could lead to potential deadlock.

Thanks!